### PR TITLE
17534 - New NSF Resource Implementation and NSF Invoice PDF Download

### PR DESCRIPTION
**Pull request by rodrigo-barraza**
*Issue #:17534 *
https://github.com/bcgov/entity/issues/17534 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/account-freeze/AccountOverview.vue [hunk 1]
```diff
@@ -85,56 +84,57 @@ import { AccessType } from '@/util/constants'
...
+  name: 'AccountOverviewView',
+  emits: ['step-forward'],
+  setup (_, { emit }) {
+    const orgStore = useOrgStore()
+    const currentOrganization: Organization = orgStore.currentOrganization
+    const currentMembership: Member = orgStore.currentMembership
+    const calculateFailedInvoices: any = orgStore.calculateFailedInvoices
+    const downloadNSFInvoicesPDF: any = orgStore.downloadNSFInvoicesPDF
+    const formatDate = CommonUtils.formatDisplayDate
+    const nsfFee = ref<number>(0)
```
Comment (seeker25) [inline auth-web/src/components/auth/account-freeze/AccountOverview.vue:104]: please wrap these in a reactive and use toRefs

**Comment by ochiu** [comment]
@rodrigo-barraza looks like we might need another version bump

File: auth-web/src/components/auth/account-freeze/AccountOverview.vue [hunk 2]
```diff
@@ -85,56 +84,53 @@ import { AccessType } from '@/util/constants'
...
+  name: 'AccountOverviewView',
+  emits: ['step-forward'],
+  setup (_, { emit }) {
+    const orgStore = useOrgStore()
+    const currentOrganization: Organization = orgStore.currentOrganization
+    const currentMembership: Member = orgStore.currentMembership
+    const calculateFailedInvoices: any = orgStore.calculateFailedInvoices
+    const downloadNSFInvoicesPDF: any = orgStore.downloadNSFInvoicesPDF
+    const formatDate = CommonUtils.formatDisplayDate
+    const suspendedDate = (currentOrganization?.suspendedOn) ? formatDate(new Date(currentOrganization.suspendedOn)) : ''
```
Comment (seeker25) [inline auth-web/src/components/auth/account-freeze/AccountOverview.vue:103]: You're going to run into date issues with timezones here, suggestion use Moment instead I think it's already included

Comment (seeker25) [reply auth-web/src/components/auth/account-freeze/AccountOverview.vue:103]: `moment(currentOrganization.suspendedOn)`

File: auth-web/src/components/auth/account-freeze/AccountOverview.vue [hunk 3]
```diff
@@ -85,56 +84,53 @@ import { AccessType } from '@/util/constants'
...
-  private nsfCount: number = 0
-  private totalTransactionAmount: number = 0
-  private totalAmountToPay: number = 0
-  private totalPaidAmount: number = 0
+export default defineComponent({
+  name: 'AccountOverviewView',
+  emits: ['step-forward'],
+  setup (_, { emit }) {
+    const orgStore = useOrgStore()
+    const currentOrganization: Organization = orgStore.currentOrganization
```
Comment (seeker25) [inline auth-web/src/components/auth/account-freeze/AccountOverview.vue:98]: are these reactive? or does it not matter?

**Comment by seeker25** [comment]
package.json and package-lock update, then we'll merge

**Review by seeker25 — APPROVED** [review]

**Pull request merged**